platform-hub-api/app/services/kubernetes/token_lookup.rb

```diff
@@ -0,0 +1,56 @@
+module Kubernetes
+  module TokenLookup
+
+    def lookup(token, kind = 'user')
+      lookup_static_tokens(token, kind) || lookup_identities(token)
+    end
+
+    def lookup_static_tokens(token, kind = 'user')
+      clusters = HashRecord.kubernetes.find_by(id: 'clusters')
+      return nil if clusters.blank?
+
+      clusters.data.each do |cluster|
+        static_user_tokens = HashRecord.kubernetes.find_by(id: "#{cluster['id'].to_s}-static-#{kind.to_s}-tokens")
+        next if static_user_tokens.blank?
+
+        existing_token = static_user_tokens.data.find {|t| ENCRYPTOR.decrypt(t['token']) == token }
+        if existing_token
+          return Hashie::Mash.new(
+            cluster: cluster['id'],
+            data: {
+              token: existing_token['token'],
+              uid: existing_token['uid'],
+              groups: existing_token['groups'],
+            },
+            kind: kind,
+            user: nil
+          )
+        end
+      end
+      nil
+    end
+
+    def lookup_identities(token)
+      Identity.kubernetes.all.each do |i|
```

Comment:
Same point, re: batching here.

Also, is it worth making a more specific query here to retrieve the exact record required, rather than traversing? (just thinking in terms of scalability of this method if we have 000s of users (we may not ;-))